### ISSUE TITLE
Deep Residual Learning for Image Recognition

### ISSUE THREAD

**Issue by leo-p** [leo-p/papers#27]
https://arxiv.org/pdf/1512.03385.pdf

Deeper neural networks are more difficult to train. We present a residual learning framework to ease the training of networks that are substantially deeper than those used previously. We explicitly reformulate the layers as learning residual functions with reference to the layer inputs, instead of learning unreferenced functions. We provide comprehensive empirical evidence showing that these residual networks are easier to optimize, and can gain accuracy from considerably increased depth. On the ImageNet dataset we evaluate residual nets with a depth of up to 152 layers---8x deeper than VGG nets but still having lower complexity. An ensemble of these residual nets achieves 3.57% error on the ImageNet test set. This result won the 1st place on the ILSVRC 2015 classification task. We also present analysis on CIFAR-10 with 100 and 1000 layers. 
The depth of representations is of central importance for many visual recognition tasks. Solely due to our extremely deep representations, we obtain a 28% relative improvement on the COCO object detection dataset. Deep residual nets are foundations of our submissions to ILSVRC & COCO 2015 competitions, where we also won the 1st places on the tasks of ImageNet detection, ImageNet localization, COCO detection, and COCO segmentation.

**Comment by leo-p** [comment]
## Summary:
- *Dataset:* [CIFAR10](https://www.cs.toronto.edu/~kriz/cifar.html), [PASCAL](http://host.robots.ox.ac.uk/pascal/VOC/) and [COCO](http://mscoco.org/).
- *Objective:* Solve the degradation problem where adding layers induces a higher training error.

## Inner-workings:
They argue that it is easier to learn the difference to the identity (the residual) than the actual mapping. Basically they start with the identity and learn the residual mapping.
This allows for easier training and thus deeper network.

## Architecture:
They introduce two new building block for Residual Networks, depending on the input dimensionality:
<img width="200" alt="screen shot 2017-05-31 at 3 49 59 pm" src="https://cloud.githubusercontent.com/assets/17261080/26635061/d489dbe2-4618-11e7-911e-68772265ee9f.png">
<img width="300" alt="screen shot 2017-05-31 at 3 57 47 pm" src="https://cloud.githubusercontent.com/assets/17261080/26635420/f6f22af8-4619-11e7-9639-ed651f8b18bb.png">

That can then be chained to produce network such as:
<img width="312" alt="screen shot 2017-05-31 at 3 54 16 pm" src="https://cloud.githubusercontent.com/assets/17261080/26635258/7b64530c-4619-11e7-81c8-5d6be547da77.png">

## Results:
Won most 1st places, very impressive and adding layers do increase accuracy.
